optimum/onnx/transformations_utils.py

```diff
@@ -45,13 +46,19 @@ def _find_duplicate_initializers(
                     if tensor_data:
                         tensor_data = tuple(tensor_data)
                         break
-                duplicates[(initializer.data_type, tensor_data, tensor_dims)].add((initializer.name, i))
+
+                # Hash tensor data to avoid storing large amounts of data in memory
+                hashed = hashlib.sha512()
+                hashed.update(str(tensor_data).encode("utf-8"))
```

Comment:
Here's a proposal that seem to work fine and is much faster:
```suggestion
                hashed.update(numpy_helper.to_array(initializer))
```

Could you give it a try?